src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -2061,17 +2061,31 @@ CommandLineInterface::InterpretArgument(const std::string& name,
 #endif  // _WIN32
 
   } else if (name == "-I" || name == "--proto_path") {
+    std::string value_split = value;
+#if defined(__CYGWIN__) || defined(__MSYS_) || defined(__MSYS2__)
+    if (IsWindowsAbsolutePath(value_split)) {
```

Comment:
How about making it so that if the value appears to start with a Windows absolute path, then we just use semicolon as the path separator? I think that would simplify the logic and hopefully make things work the way users would expect.

Reply:
@acozzette , I believe the follow commit hash reference link addresses this concern, as well as affords more flexibility around lists of absolute paths provided as arguments...

https://github.com/PierceLBrooks/protobuf/pull/21244/commits/755a37f99e3803465b082532018faa863b0317e9